akka-actor-typed-tests/src/test/scala/akka/actor/typed/pubsub/LocalPubSubSpec.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.pubsub
+
+import akka.actor.testkit.typed.scaladsl.LoggingTestKit
+import akka.actor.testkit.typed.scaladsl.LogCapturing
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import org.scalatest.wordspec.AnyWordSpecLike
+import scala.concurrent.duration._
+
+class LocalPubSubSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike with LogCapturing {
+
+  "A pub-sub topic running locally" must {
+
+    "publish to all local subscriber actors of a topic" in {
+      val fruitTopic =
+        LoggingTestKit.debug("Topic list updated").expect {
+          testKit.spawn(Topic[String]("fruit"))
+        }
+
+      try {
+        val probe1 = testKit.createTestProbe[String]
+        val probe2 = testKit.createTestProbe[String]
+        val probe3 = testKit.createTestProbe[String]
+
+        LoggingTestKit.debug("Topic list updated").expect {
+          fruitTopic ! Topic.Subscribe(probe1.ref)
+          fruitTopic ! Topic.Subscribe(probe2.ref)
+          fruitTopic ! Topic.Subscribe(probe3.ref)
+        }
+
+        fruitTopic ! Topic.Publish("banana")
+        probe1.expectMessage("banana")
+        probe2.expectMessage("banana")
+        probe3.expectMessage("banana")
+
+      } finally {
+        testKit.stop(fruitTopic)
+      }
+    }
+
+    "publish to all subscriber actors across several instances of the same topic" in {
+      val fruitTopic1 =
+        LoggingTestKit.debug("Topic list updated").expect {
+          testKit.spawn(Topic[String]("fruit"))
+        }
+      val fruitTopic2 =
+        LoggingTestKit.debug("Topic list updated").expect {
+          testKit.spawn(Topic[String]("fruit"))
+        }
+
+      try {
+        val probe1 = testKit.createTestProbe[String]
```

Comment:
will you get told off for this in scala 3? Add ()?

Reply:
Makes sense anyways because it side-effects. 👍